src/pdc_osti/poster.py

```diff
@@ -255,35 +248,51 @@ def run_pipeline(self) -> None:
 
 
 def main() -> None:
-    log = script_log_init(SCRIPT_NAME)
-    args = sys.argv
+    parser = argparse.ArgumentParser(
+        description="Script to post new Princeton datasets to DOE/OSTI"
+    )
+    parser.add_argument(
+        "-s",
+        "--source",
+        required=False,
+        default="",
+        type=str,
+        help="Source for Princeton data (dspace or pdc)",
+    )
+    parser.add_argument(
+        "-m",
+        "--mode",
+        default="dry-run",
+        type=str,
+        help="Mode of KPI operation (dry-run, test, or execute)",
+    )
+    args = parser.parse_args()
```

Comment:
Changes look good.

We will need to handle the minted DOI (#40) and ORCID identifiers (no ticket yet) separately.